fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ConcreteAvailability.java

```diff
@@ -64,43 +42,30 @@ public SimplifiedIntervalList getAvailableIntervals(PhysicalDataSourceConstraint
                 .map(physicalName -> getAllAvailableIntervals().getOrDefault(
                         physicalName,
                         new SimplifiedIntervalList()
-                )).reduce(SimplifiedIntervalList::intersect).orElse(new SimplifiedIntervalList());
-    }
-
-
-    /**
-     * Returns the name of the table and data source associated with this Availability.
-     *
-     * @return the name of the table and data source associated with this Availability
-     */
-    protected TableName getName() {
-        return name;
+                ))
+                .reduce(SimplifiedIntervalList::intersect)
+                .orElse(new SimplifiedIntervalList());
     }
 
     @Override
     public String toString() {
-        return String.format(
-                "ConcreteAvailability for table: %s with data source names %s",
-                name.asName(),
-                dataSourceNames
-        );
+        return String.format("ConcreteAvailability for data source: %s", getDataSourceName().asName());
     }
 
     @Override
     public boolean equals(Object obj) {
         if (obj instanceof ConcreteAvailability) {
             ConcreteAvailability that = (ConcreteAvailability) obj;
-            return Objects.equals(name.asName(), that.name.asName())
-                    && Objects.equals(dataSourceNames, that.dataSourceNames)
+            return Objects.equals(getDataSourceName().asName(), that.getDataSourceName().asName())
                     // Since metadata service is mutable, use instance equality to ensure table equality is stable
-                    && metadataService == that.metadataService;
+                    && getDataSourceMetadataService() == that.getDataSourceMetadataService();
         }
```

Comment:
The type matters (ie. a concrete shouldn't be considered the same as a permissive) because they have different behavior. _Some_ of this can be moved up to the parent, but not fully I don't think... Unless I'm wrong?